projects/fw_103/src/main.c

```diff
@@ -0,0 +1,43 @@
+#include <stdbool.h>
+
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+void readADC(const GpioAddress *btn_addr_ptr, void *context) {
```

Comment:
For functions that are only used in the scope of the file, we make them private (ie declare them static). We also prefix the function name with `prv_` to let everyone know that they should only be used here. 

Nit: We use the convention of lower case in function prototypes and variables with underscores (ie `prv_read_adc` instead.
Structs and types generally use Camel case (GpioAddress, etc)